src/Tests/Microsoft.NET.Sdk.BlazorWebAssembly.AoT.Tests/WasmAoTPublishIntegrationTest.cs

```diff
@@ -0,0 +1,190 @@
+            // referenced by the Hosted project appear in the publish directory
+            publishDirectory.Should().HaveFiles(new[]
+            {
+                "RazorClassLibrary.dll",
+                "blazorwasm.dll"
+            });
+
+            publishDirectory.Should().HaveFiles(new[]
+            {
+                "wwwroot/_framework/blazor.boot.json",
+                "wwwroot/_framework/blazor.webassembly.js",
+                "wwwroot/_framework/dotnet.wasm",
+                "wwwroot/_framework/blazorwasm.dll",
+                "wwwroot/_framework/System.Text.Json.dll"
+            });
+
+            // Verify project references appear as static web assets
+            // Also verify project references to the server project appear in the publish output
+            publishDirectory.Should().HaveFiles(new[]
+            {
+                "wwwroot/_framework/RazorClassLibrary.dll",
+                "RazorClassLibrary.dll"
+            });
+
+            // Verify static assets are in the publish directory
+            publishDirectory.Should().HaveFiles(new[]
+            {
+                "wwwroot/index.html"
+            });
+
+            // Verify scoped css
+            publishDirectory.Should().HaveFiles(new[]
+            {
+                "wwwroot/blazorwasm.styles.css"
+            });
+
+            // Verify static web assets from referenced projects are copied.
+            publishDirectory.Should().HaveFiles(new[]
+            {
+                "wwwroot/_content/RazorClassLibrary/wwwroot/exampleJsInterop.js",
+                "wwwroot/_content/RazorClassLibrary/styles.css",
+            });
+
+            // Verify web.config
+            publishDirectory.Should().HaveFiles(new[]
+            {
+                "web.config"
+            });
+
+            VerifyBootManifestHashes(testInstance, Path.Combine(publishDirectory.ToString(), "wwwroot"));
+
+            // Verify compression works
+            publishDirectory.Should().HaveFiles(new[]
+            {
+                "wwwroot/_framework/dotnet.wasm.br",
+                "wwwroot/_framework/blazorwasm.dll.br",
+                "wwwroot/_framework/RazorClassLibrary.dll.br",
+                "wwwroot/_framework/System.Text.Json.dll.br"
+            });
+
+            VerifyBootManifestHashes(testInstance, blazorPublishDirectory);
+            VerifyServiceWorkerFiles(testInstance, blazorPublishDirectory,
+                serviceWorkerPath: Path.Combine("serviceworkers", "my-service-worker.js"),
+                serviceWorkerContent: "// This is the production service worker",
+                assetsManifestPath: "custom-service-worker-assets.js");
+        }
+
+        private TestAsset CreateAspNetSdkTestAssetWithAot(
+            string testAsset,
+            string[] projectsToAoT,
+            [CallerMemberName] string callerName = "") =>
+            CreateAspNetSdkTestAsset(testAsset, callerName: callerName, identifier: "AoT")
```

Comment:
I'm personally not a fan of multi-line lambda expressions. It sits in the poor readibility bucket with the risk of having to write more code the moment you have to do something like perform an extra check or transform an argument.